app/src/main/java/io/gnosis/safe/ui/safe/add/AddSafeFragment.kt

```diff
@@ -45,12 +50,15 @@ class AddSafeFragment : BaseViewBindingFragment<FragmentAddSafeBinding>() {
         super.onViewCreated(view, savedInstanceState)
         with(binding) {
             nextButton.setOnClickListener {
-                addSafeAddressInputLayout.address?.let { viewModel.validate(it) }
+                addSafeAddressInputLayout.address?.let { viewModel.validate(it, selectedChain) }
             }
             backButton.setOnClickListener { findNavController().navigateUp() }
             addSafeAddressInputLayout.setOnClickListener {
                 addressInputHelper.showDialog()
             }
+            chainRibbon.text = selectedChain.name
+            chainRibbon.setBackgroundColor(Color.parseColor(selectedChain.backgroundColor))
```

Comment:
I would use try catch here (see applying colors for ribbon in StartActivity) as color parsing can fail due to possible malformed format coming from backend